txpipe/covariance.py

```diff
@@ -61,206 +77,419 @@ def read_cosmology(self):
+                        tracer_Noise=tracer_Noise,
+                        two_point_data=two_point_data,
+                        cache=cl_cache,
+                        WT=WT,
+                    )
+
+                # Fill in this chunk of the matrix
+                cov_ij = cov_ij['final_b']
+                # Find the right location in the matrix
+                start_i = i * Nell_bins
+                start_j = j * Nell_bins
+                end_i = start_i + Nell_bins
+                end_j = start_j + Nell_bins
+                # and fill it in, and the transpose component
+                cov_full[start_i:end_i, start_j:end_j] = cov_ij
+                cov_full[start_j:end_j, start_i:end_i] = cov_ij.T
+
+        try:
+            np.linalg.cholesky(cov_full)
+        except:        
+            print("liAnalg.LinAlgError: Covariance not positive definite! "
+                "Most likely this is a problem in xim. "
+                "We will continue for now but this needs to be fixed.")
+
+        return cov_full
+
+
+class TXRealGaussianCovariance(TXFourierGaussianCovariance):
+    name='TXRealGaussianCovariance'
+    do_xi = True
 
-                print(" - Calculating fiducial C_ell for ", data_type, t1, t2)
-                theory[(data_type, t1, t2)] = ccl.angular_cl(cosmo, cTracer1, cTracer2, ell)
+    inputs = [
+        ('fiducial_cosmology', YamlFile),     # For the cosmological parameters
+        ('photoz_stack', HDFFile),            # For the n(z)
+        ('twopoint_data_real', SACCFile),     # For the binning information
+        ('tracer_metadata', HDFFile),         # For metadata
 
-        return theory
+    ]
 
+    outputs = [
+        ('summary_statistics_real', SACCFile),
+    ]
 
-    def compute_noise_c_ell(self, n_eff, sigma_e, n_lens, sacc_data):
-        import sacc
+    config_options = {
+        'min_sep':2.5,  # arcmin
+        'max_sep':250,
+        'nbins':20,
+            }
 
-        CEE=sacc.standard_types.galaxy_shear_cl_ee
-        CEd=sacc.standard_types.galaxy_shearDensity_cl_e
-        Cdd=sacc.standard_types.galaxy_density_cl
-
-        noise = {}
-
-        for data_type in sacc_data.get_data_types():
-            for (tracer1, tracer2) in sacc_data.get_tracer_combinations(data_type):
-                ell = sacc_data.get_tag('ell', data_type, (tracer1, tracer2))
-                n = len(ell)
-                b = int(tracer1.split("_")[1])
-                if tracer1 != tracer2:
-                    N = np.zeros_like(ell)
-                elif data_type == CEE:
-                    N = np.ones(n) * (sigma_e[b]**2 / n_eff[b])
-                else:
-                    assert data_type == Cdd
-                    N = np.ones(n) / n_lens[b]
-                noise[(data_type, tracer1, tracer2)] = N
+    def run(self):
+        super().run()
 
-        return noise
+    def get_angular_bins(self, two_point_data):
+        # this should be changed to read from sacc file
```

Comment:
you're right these aren't in twopoint.py yet - will add them.

Reply:
can leave this comment here for now

Reply:
ok